icu4c/source/common/loclikely.cpp

```diff
@@ -961,17 +965,17 @@ _uloc_minimizeSubtags(const char* localeID,
     {
         icu::CharString base;
         {
-            icu::CharStringByteSink sink(&base);
+            icu::CharStringByteSink baseSink(&base);
```

Comment:
See previous comment about consistenly using the same name to refer to the same object.

If I understand your PR description correctly, you think that it might cause confustion that it in two different scopes within this function exist two different objects that both are named `sink`.

Is that really a problem? We have lots and lots for for-loops that all name their loop variable `i`, even though they each have their own separate locally scoped loop variable. That doesn't seem to cause any confusion. Is the situation with the two Sink objects here really different from that?

Reply:
I think it is very confusing. Would you write a function like tis
void Foo(int i) {
 // 50 lines of code
 for (int i = 0; ...) {
     b(i);
 }
 // 50 lines of code
 for (int i = 0; ...) {
     b(i);
 }
 // 50 lines of code
 for (int i = 0; ...) {
     b(i);
 }
 // 50 lines of code
 b(i);
}
Basically that is what happen here. The name "sink" got passed in, and then some other local sink are declare which "shadow" the sink in the parameter for 4-5 times, and then later the real "sink" got called. It is very difficult to track WHICH sink it is at the point of the last call and I think that could easily introduce bug

Reply:
I think renaming the inner sinks is fine.